### PR TITLE
Migrate/vitepress

### PR DESCRIPTION
**Pull request by TerminalFi**
Github Actions untested.

### CODE REVIEW

**Comment by FichteFoll** [comment]
Thanks for your work on this. I tried it out briefly locally. 

Here's some feedback:

- Not digging the font at all. It's too playful and thin.
- Glossary link from the landing page doesn't work. Not from the nav bar either.
- I like the simple light/dark theme switch (we had arguably too much choice before tbh), but the orange accent on the light theme has too little contrast.
- In the old vuepress theme, the active section (guide/reference) in the navbar was highlighted even when not directly on the landing page. It's missing in the vitepress version, but not that important.
- The heading of the contributors is cut off for me (FireFox). [2023-10-24_20-47-59](https://github.com/sublimetext-io/docs.sublimetext.io/assets/931051/ced4dbab-b926-403f-868c-f326d307f541)
- There's a bit too much spacing around the contributors. In order to still have it fit on the page width-wise, maybe we can show 10 by default? Or even more and make them smaller at the same time because I do believe it takes quite a large portion of the screen.
- I think we can omit the `docs:` prefix of our scripts when the docs are the main focus of this project.
- With JS disabled, switching between pages briefly flashes the theme from dark to light, though I assume that's a thing from the default theme.
- The contributors component also doesn't function without JS, but that's okay for now, although it *could* technically function without.
- With the section parent nodes not requiring a page anymore, we can remove a couple of the "Overview" pages (in particular Customization and Extensibility).

Outside of that, I like the enhancements that vitepress brings, for example the page nagivation on the right side, and I like that we're not using the custom theme anymore because that could've become a maintenance burden in the future.

Also, when merging I'd like to squash the commits to preserve some of the history between the old .vuepress and the new .vitepress folder.

File: .github/workflows/gh-pages.yml [hunk 1]
```diff
@@ -17,17 +17,20 @@ jobs:
...
       - name: Install Dependencies
-        run: yarn install
+        run: pnpm install
 
       - name: Build Docs
-        run: yarn build
+        run: pnpm run dev:build
 
       - name: Deploy Docs
         uses: peaceiris/actions-gh-pages@v3
```
[the remark this answers is not in the document]
Comment (TerminalFi) [reply .github/workflows/gh-pages.yml:32]: I believe that is a different GH action. Here is the one we use https://github.com/peaceiris/actions-gh-pages/releases

File: docs/.vitepress/components/Contributors.vue [hunk 2]
```diff
@@ -0,0 +1,107 @@
+<template>
+  <div>
+    <h1 v-if="!error">Contributors ({{ contributors.length }})</h1>
+    <VPTeamMembers v-if="!error" size="small" :members="visibleContributors" />
+    <div v-if="moreContributors.length">
+      <div v-show="showMore">
+        <VPTeamMembers size="small" :members="moreContributors" />
+      </div>
+      <button @click="showMore = !showMore">
```
[the remark this answers is not in the document]
Comment (TerminalFi) [reply docs/.vitepress/components/Contributors.vue]: Didn't seem to work well with svg's

**Comment by TerminalFi** [comment]
> * In the old vuepress theme, the active section (guide/reference) in the navbar was highlighted even when not directly on the landing page. It's missing in the vitepress version, but not that important.

Can you share a screenshot?

> Glossary link from the landing page doesn't work. Not from the nav bar either.

Working for me. What is the link value when hovering over either link?

**Comment by TerminalFi** [comment]
> The contributors component also doesn't function without JS, but that's okay for now, although it could technically function without.

Can you test now?

> There's a bit too much spacing around the contributors. In order to still have it fit on the page width-wise, maybe we can show 10 by default? Or even more and make them smaller at the same time because I do believe it takes quite a large portion of the screen.

How is that?

**Comment by TerminalFi** [comment]
@jrappen 

Maybe you can see an issue. Search disappeared, even though it is configured

**Comment by FichteFoll** [comment]
I pushed some changes, please take a look. Looking pretty good so far and I would argue content-wise it's ready to merge, but I'd like to mingle the commits a bit before doing that.

File: package.json [hunk 3]
```diff
@@ -1,19 +1,19 @@
...
-    "vue-click-outside": "^1.1.0",
-    "vuepress": "^1.9.9",
-    "vuepress-plugin-container": "^2.1.2",
+    "vitepress": "1.0.0-rc.24",
+    "vue": "^3.3.7",
     "vuepress-plugin-glossary": "^1.0.2"
-  },
-  "devDependencies": {
-    "vuepress-plugin-check-md": "^0.0.3"
   }
```
[the remark this answers is not in the document]
Comment (FichteFoll) [reply package.json:22]: What's the effect of pinning the pnpm version here? it didn't seem to complain when I was using it with 8.9.2 at least.

**Comment by FichteFoll** [comment]
I have rebased it on top of master and merged a couple commits with follow-up fixes (and specifically also the "remove vuepress" and "add vitepress" ones because having these separate made no sense).

There are still some problems, but let's proceed with this for now and create issue for further follow-ups.

**Review by FichteFoll — APPROVED** [review]

**Pull request merged**